addons/website/static/src/js/editor/widget_link.js

```diff
@@ -38,39 +43,81 @@ weWidgets.LinkTools.include({
             classes: {
                 "ui-autocomplete": 'o_website_ui_autocomplete'
             },
-        }
-        wUtils.autocompleteWithPages(this, this.$('input[name="url"]'), options);
+        };
+        this.$urlInput = this.$('input[name="url"]');
+        $('.o_we_link_action_change').toggleClass('d-none', !this.$urlInput.val().includes('/web/'));
+        wUtils.autocompleteWithPages(this, this.$urlInput, options);
         this._adaptPageAnchor();
         return def;
     },
+    isDownloadEnable: function () {
+        return this.$link.attr('data-link-action') === 'download';
+    },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * @override
+     */
+    _isNewWindow: function (url) {
+        // If URL starts with /web/ or if the file name is included in URL, then it's a file.
+        // Thus, we want to open it in a new window.
+        const fileName = this.$link.attr('data-file-name') && this.$link.attr('data-file-name').replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+        const isFileLink = new RegExp(`${fileName ? fileName + '|' : ''}\\/web\\/`).test(url);
+        // If a link that pointed to a file before doesn't anymore,
+        // Or if a link that didn't point to a file does now, we need to update the toolbar.
+        $('.o_we_link_replace_file').parent().toggle(isFileLink);
+        $('.o_we_link_upload_file').toggle(!isFileLink);
```

Comment:
The method name starts with "_is" meaning it should be a method which returns a boolean and does not modify anything... but here it toggles stuff